File: NetCoreApi2/Data/Models/Clientes.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace NetCoreApi2.Data.Models
{
    public class ClientesModel
    {
        [Key]
        public int Id { get; set; }
        public string Codigo { get; set; }
        public string Nombre { get; set; }
        public int EmpresaId { get; set; }
        public EmpresasModel empresa { get; set; }
        public ICollection<ProyectosModel> Proyectos { get; set; }
    }
}

File: NetCoreApi2/clientApp/src/features/clientes/models/clientes.ts
import {ProyectosModel} from "@/features/proyectos/models/proyectos.ts";

export interface ClientesInterface {
    id: number;
    codigo: string;
    nombre: string;
    empresasId: number;
    proyectos: ProyectosModel[];
}//end interface

export class ClientesModel{
    id: number;
    codigo: string;
    nombre: string;
    empresaId: number;
    proyectos: ProyectosModel[];

    constructor(data: ClientesInterface){
        this.id = data.id;
        this.codigo = data.codigo;
        this.nombre = data.nombre;
        this.empresaId = data.empresasId;
        this.proyectos = data.proyectos;
    }//end constructor
}//end class Clientes